source_code/main_mcu/src/GUI/gui_prompts.c

```diff
@@ -1115,6 +1136,7 @@ RET_TYPE gui_prompts_get_user_pin(volatile uint16_t* pin_code, uint16_t stringID
     #endif
     
     BOOL random_pin_feature_enabled = (BOOL)custom_fs_settings_get_device_setting(SETTING_RANDOM_PIN_ID);
+    BOOL show_pin_entry = ((BOOL)custom_fs_settings_get_device_setting(SETTINGS_PARANOID_PIN_ENTRY) == TRUE) ? FALSE : TRUE;
```

Comment:
I should have caught that earlier... but could you do a != FALSE instead, as that value will be programmed by moolticute?
sorry about this additional request :)

Reply:
No worries!
Indeed it will more relevant to have this condition the other way around :)